plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -327,6 +355,62 @@ func (s *SQLServer) accRow(query Query, acc telegraf.Accumulator, row scanner) e
 	return nil
 }
 
+// gatherHealth stores info about any query errors in the healthMetrics map
+func (s *SQLServer) gatherHealth(healthMetrics map[string]*HealthMetric, serv string, queryError error, mutex *sync.Mutex) {
+	if !s.HealthMetric {
+		return
+	}
+
+	mutex.Lock()
+
+	if healthMetrics[serv] == nil {
+		healthMetrics[serv] = &HealthMetric{}
+	}
+
+	healthMetrics[serv].AttemptedQueries++
+	if queryError == nil {
+		healthMetrics[serv].SuccessfulQueries++
+	}
+
+	mutex.Unlock()
+}
+
+// accHealth accumulates the query health data contained within the healthMetrics map
+func (s *SQLServer) accHealth(healthMetrics map[string]*HealthMetric, acc telegraf.Accumulator) {
+	if !s.HealthMetric {
+		return
+	}
```

Comment:
Same as above.

Reply:
See my response above.

Reply:
As I said above, we might want to do this one level up. Furthermore, you don't need this here at all (besides for documentation) as the gather method above only fills the `healthMetrics` map in case the condition is true. So the loop below will not be run.

Reply:
The way you suggested below works. I'll go with that. 

I think keeping the `if s.HealthMetric` on line 274 helps with readability. It looks visually similar to the same check on line 261, so it's clear they are related bits of code.